src/sagemaker/session.py

```diff
@@ -124,11 +125,12 @@ def upload_data(self, path, bucket=None, key_prefix='data'):
             key_suffix = name
 
         bucket = bucket or self.default_bucket()
-        s3 = self.boto_session.resource('s3')
+        s3_client = self.boto_session.client('s3')
```

Comment:
Could you describe what the exact difference is between this 2 approaches for uploading files to `s3`?

Reply:
The [transfer manager](http://boto3.readthedocs.io/en/latest/guide/s3.html?highlight=s3transfer#using-the-transfer-manager) adds functionality to the SDK that's not strictly part of the REST API. A couple of those things are uploading directly from files (notice in the diff that before they were reading in bytes and `.put()`ing those directly to the object in S3) and multi-part uploads. Without that functionality we would have to manage the multi-part upload on our own.

But actually it looks like all the `upload_file()` methods work the same under the hood. So I can use the `s3.Object` that they're instantiating already instead of using `S3Transfer` directly.